framework/Samplers/MCMC/MCMC.py

```diff
@@ -214,32 +262,40 @@ def initialize(self, externalSeeding=None, solutionExport=None):
     for _, dist in self._availProposal.items():
       dist.initializeDistribution()
     self._acceptDist.initializeDistribution()
-    for var, dist in self._proposal.items():
-      if dist:
-        self._proposal[var] = self.retrieveObjectFromAssemblerDict('proposal', dist)
+    for var in self._updateValues:
+      if var in self._proposal:
+        self._proposal[var] = self.retrieveObjectFromAssemblerDict('proposal', self._proposal[var])
         distType = self._proposal[var].getDistType()
         if distType != 'Continuous':
           self.raiseAnError(IOError, 'variable "{}" requires continuous proposal distribution, but "{}" is provided!'.format(var, distType))
       else:
         self._proposal[var] = self._availProposal['normal']
+
     AdaptiveSampler.initialize(self, externalSeeding=externalSeeding, solutionExport=solutionExport)
     ## TODO: currently AdaptiveSampler is still using self.assemblerDict to retrieve the target evaluation.
     # We should change it to using the following method.
     # retrieve target evaluation
     # self._targetEvaluation = self.retrieveObjectFromAssemblerDict('TargetEvaluation', self._targetEvaluation)
 
-    # initialize the input variable values
-    for key, value in self._updateValues.items():
-      totDim = self.variables2distributionsMapping[key]['totDim']
-      dist = self.distDict[key]
+    for var, priorFun in self._priorFuns.items():
+      self._priorFuns[var] = self.retrieveObjectFromAssemblerDict('probabilityFunction', priorFun)
+      if "pdf" not in self._priorFuns[var].availableMethods():
```

Comment:
you can ask the Function to check (At the initialization stage) to check if a method is present. In that way you will not need to do the check here since the code will error out right at the begin

Reply:
The requested capability is not available within Function. For now, we will keep the check here.